dashboard/app/controllers/levels_controller.rb

```diff
@@ -203,6 +204,10 @@ def embed_level
     render 'levels/show'
   end
 
+  def set_applab_user_id
+    @applab_user_id = applab_user_id
```

Comment:
since applab_user_id doesn't do anything it's hard to tell what the implementation should be.. but I think if you're worried about calling it unnecessarily, the right logic to decide when it should be called might not be "are we LevelsController/show or edit". Eg. do you need the applab_user_id if it's not an applab level...